src/assertions/similar.ts

```diff
@@ -1,18 +1,20 @@
 import invariant from 'tiny-invariant';
 import { matchesSimilarity } from '../matchers';
 import type { Assertion, AssertionValue, AtomicTestCase, GradingResult } from '../types';
+import { coerceString } from './utils';
 
 export const handleSimilar = async (
   assertion: Assertion,
   renderedValue: AssertionValue | undefined,
-  outputString: string,
+  output: string | object,
   inverse: boolean,
   test: AtomicTestCase,
 ): Promise<GradingResult> => {
   invariant(
     typeof renderedValue === 'string' || Array.isArray(renderedValue),
     'Similarity assertion type must have a string or array of strings value',
   );
+  const outputString = coerceString(output);
```

Comment:
instead of having to remember to call coerceString everywhere, it might makes sense to pass `outputString` into the handlers.  the handler signature is large enough that a dict {} would be less unwieldy